cli/command/trust/helpers.go

```diff
@@ -25,3 +29,50 @@ func getImageReferencesAndAuth(cli command.Cli, imgName string) (reference.Named
 	authConfig := command.ResolveAuthConfig(ctx, cli, repoInfo.Index)
 	return ref, repoInfo, &authConfig, err
 }
+
+// GetSignableRoles returns a list of roles for which we have valid signing
+// keys, given a notary repository and a target
+func GetSignableRoles(repo *client.NotaryRepository, target *client.Target) ([]data.RoleName, error) {
```

Comment:
as mentioned in sync: we could further refactor this with an unexported inner function that takes the output of `repo.CryptoService.ListAllKeys()` and `repo.GetDelegationRoles()` as arguments. This will allow us to properly unit test